packages/tpetra/core/example/Finite-Element-Assembly/fem_assembly_LocalElementLoop_DP.hpp

```diff
@@ -174,15 +180,22 @@ int executeLocalElementLoopDP_(const comm_ptr_t& comm, const struct CmdLineOpts&
     //
     for(size_t element_node_idx=0; element_node_idx<owned_element_to_node_ids.extent(1); element_node_idx++)
     {
-       crs_graph_overlapping->insertGlobalIndices(global_ids_in_row[element_node_idx], global_ids_in_row());
+      if( mesh.nodeIsOwned(global_ids_in_row[element_node_idx]) )
```

Comment:
What does `element_node_idx` actually return?  Is that an element index?  If so, then this is all wrong -- you would need a second loop over rows of the element.

Reply:
If it's wrong would it be generating a bad mesh or would it do something more subtle?

Reply:
@mhoemmen 
`element_node_idx` is an index into the `owned_element_to_node_ids` array.  It is not a global node index. Accessing `global_ids_in_row[element_node_idx]` gives me the global node id of my node.

for example, for a 1 process 3x3 test: "[pid] : (element_node_idx, global_ids_in_row[element_node_idx])":

```
element 0
[0] : (0 ,0)     mesh.nodeIsOwned( 0 ) => true
[0] : (1 ,1)     mesh.nodeIsOwned( 1 ) => true
[0] : (2 ,5)     mesh.nodeIsOwned( 5 ) => true
[0] : (3 ,4)     mesh.nodeIsOwned( 4 ) => true
element 1
[0] : (0 ,1)    .
[0] : (1 ,2)    .
[0] : (2 ,6)    .
[0] : (3 ,5)    .
element 2
[0] : (0 ,2)
[0] : (1 ,3)
[0] : (2 ,7)
[0] : (3 ,6)
element 3
[0] : (0 ,4)
[0] : (1 ,5)
[0] : (2 ,9)
[0] : (3 ,8)
element 4
[0] : (0 ,5)
[0] : (1 ,6)
[0] : (2 ,10)
[0] : (3 ,9)
element 5
[0] : (0 ,6)
[0] : (1 ,7)
[0] : (2 ,11)
[0] : (3 ,10)
element 6
[0] : (0 ,8)
[0] : (1 ,9)
[0] : (2 ,13)
[0] : (3 ,12)
element 7
[0] : (0 ,9)
[0] : (1 ,10)
[0] : (2 ,14)
[0] : (3 ,13)
element 8
[0] : (0 ,10)
[0] : (1 ,11)
[0] : (2 ,15)
[0] : (3 ,14)
```